lib/Monad_WP/NonDetMonad.thy

```diff
@@ -236,6 +236,12 @@ definition
   gets_map :: "('s \<Rightarrow> 'a \<Rightarrow> 'b option) \<Rightarrow> 'a \<Rightarrow> ('s, 'b) nondet_monad" where
   "gets_map f p \<equiv> gets f >>= (\<lambda>m. assert_opt (m p))"
 
+definition
+  fun_of_m :: "('s, 'a) nondet_monad \<Rightarrow> 's \<Rightarrow> 'a option"
```

Comment:
It looks like `fun_of_m` is not used any more, apart from in one lemma which is itself not used. Can you please remove? (It's better if there the temptation to use something like it is small)